File: examples/weather-app.ts
/**
 * Example: Weather App
 *
 * A full SwiftUI app scaffold with a main window,
 * macOS settings scene, and persisted user preferences.
 *
 * Run:
 *   axint compile examples/weather-app.ts --out generated/
 */

import { defineApp, scene, storage } from "@axint/compiler";

export default defineApp({
  name: "WeatherApp",
  scenes: [
    scene.windowGroup("WeatherDashboard"),
    scene.settings("SettingsView", { platform: "macOS" }),
  ],
  appStorage: {
    useCelsius: storage.boolean("use_celsius", true),
    lastCity: storage.string("last_city", "Cupertino"),
    refreshMinutes: storage.int("refresh_minutes", 30),
  },
});

File: python/examples/weather_app.py
"""Example: weather app scaffold.

Run it:

    axint-py parse examples/weather_app.py
    axint-py compile examples/weather_app.py --stdout
"""

from axint import define_app, scene, storage

weather_app = define_app(
    name="WeatherApp",
    scenes=[
        scene.window_group("WeatherDashboard"),
        scene.settings("SettingsView", platform="macOS"),
    ],
    app_storage={
        "use_celsius": storage.boolean("use_celsius", True),
        "last_city": storage.string("last_city", "Cupertino"),
        "refresh_minutes": storage.int("refresh_minutes", 30),
    },
)
